awslabs/aws-lambda-powertools-typescript
# chore(examples): updated contributing & docs with SAM

## Description of your changes

Now that the project offers SAM-based examples, there are some changes needed to the documentation, contributing guidelines, and readme files needed to reflect this change. This PR introduces these changes and it should be the last one related to SAM.

### List of changes:

#### Readme files

This includes:
* Project's README.md
* `packages/tracer/README.md`
* `packages/metrics/README.md`
* `packages/logger/README.md`

![image](https://user-images.githubusercontent.com/7353869/168071272-94cf9567-75c6-461a-a537-04376db957bf.png)
The "Examples" link above used to bring you to the `examples/cdk` folder, it now brings you to `examples` so you can choose your favourite flavour to use.
![image](https://user-images.githubusercontent.com/7353869/168071436-0e1546e7-9d8e-4416-b9cf-32c06eb26a5f.png)
Added the link to the SAM samples under the "Examples" section.

#### Documentation

![image](https://user-images.githubusercontent.com/7353869/168071900-672032f4-c10f-43cf-8f69-7b444dbffc42.png)

In the main page (`docs/index.md`) there is a list of items under the "Examples" section, I have added an item for SAM

#### Contributing

![image](https://user-images.githubusercontent.com/7353869/168072250-79a6475a-bf18-45bf-879c-f1337c0a1395.png)

I have added AWS SAM CLI as requirement for the project, as it's now needed in order to build the SAM sample as part of the pre-push hook. (This is similar to CDK, but since SAM CLI is not vended via NPM we must install it manually).

<img width="609" alt="image" src="https://user-images.githubusercontent.com/7353869/168329853-d5b89eb1-f450-458c-be4e-a39f8848e0ef.png">

Finally I also added the Open in GitPod button that will automatically create the setup to start contributing in a few seconds.

#### Gitpod / Codespaces

With the latest changes I have made changes also to the configurations needed to use GitHub Codespaces & Gitpod. These include the addition of SAM CLI, as well as the initialisation of both example folders. This last step is needed to successfully execute the pre-push hooks.

### How to verify this change

Check content of the CONTRIBUTING readme ([link to rendered version in this PR](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/chore/aamorosi/sam_docs/CONTRIBUTING.md#setup)), and optionally check README files + docs links to examples.

If you want to try the Gitpod setup [click here](https://gitpod.io#https://github.com/awslabs/aws-lambda-powertools-typescript/tree/chore/aamorosi/sam_docs) (it will open a new Gitpod workspace in your account using this branch).

### Related issues, RFCs

#849 
#674

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
